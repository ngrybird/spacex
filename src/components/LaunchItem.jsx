import React from "react";

class LaunchItem extends React.Component {
  render() {
    const { launch } = this.props;
    return (
      <div className="list-item">
        <div className="mission-patch">
          <img src={launch.links.mission_patch_small} alt={"mission picture"} />
        </div>
        <div className="mission-name">
          {launch.mission_name} #{launch.flight_number}
        </div>
        {launch.mission_id.length > 0 ? (
          <>
            <div className="mission-id">Mission Ids</div>
            <ul>
              {launch.mission_id.map(id => (
                <li className="data-values">{id}</li>
              ))}
            </ul>
          </>
        ) : (
          ""
        )}
        <div className="year-field">
          <span className="launch-year">Launch Year :</span>
          <span className="data-values">{launch.launch_year}</span>
        </div>
        <div className="launch-success-field">
          <span className="launch-success-label">Successful Launch :</span>
          <span className="launch-data-values">
            {launch.launch_success ? "True" : "False"}
          </span>
        </div>
        <div className="launch-success-field">
          <span className="launch-success-label">Successful Landing :</span>
          <span className="launch-data-values">{launch.launch_success}</span>
        </div>
      </div>
    );
  }
}

export default LaunchItem;
