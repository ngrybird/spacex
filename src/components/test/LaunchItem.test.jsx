import { render, fireEvent } from "@testing-library/react";
import React from "react";
import { apiResponse } from "./testData";
import LaunchItem from "../LaunchItem";

describe("<LaunchItem/>", () => {
  it("render ListItem with mission stub", () => {
    const { container, getByText } = render(
      <LaunchItem launch={apiResponse[0]} />
    );
    expect(container).toMatchSnapshot();
    const yearValue = getByText("2020");
    expect(yearValue).toBeInTheDocument();

    const missionName = getByText("Starlink 2 #87");
    expect(missionName).toBeInTheDocument();
  });
});
