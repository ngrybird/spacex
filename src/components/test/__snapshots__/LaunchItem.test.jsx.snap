// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchItem/> render ListItem with mission stub 1`] = `
<div>
  <div
    class="list-item"
  >
    <div
      class="mission-patch"
    >
      <img
        alt="mission picture"
        src="https://images2.imgbox.com/9a/96/nLppz9HW_o.png"
      />
    </div>
    <div
      class="mission-name"
    >
      Starlink 2
       #
      87
    </div>
    
    <div
      class="year-field"
    >
      <span
        class="launch-year"
      >
        Launch Year :
      </span>
      <span
        class="data-values"
      >
        2020
      </span>
    </div>
    <div
      class="launch-success-field"
    >
      <span
        class="launch-success-label"
      >
        Successful Launch :
      </span>
      <span
        class="launch-data-values"
      >
        True
      </span>
    </div>
    <div
      class="launch-success-field"
    >
      <span
        class="launch-success-label"
      >
        Successful Landing :
      </span>
      <span
        class="launch-data-values"
      />
    </div>
  </div>
</div>
`;
